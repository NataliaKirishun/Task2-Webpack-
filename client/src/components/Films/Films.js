import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Theme from '../../common/Theme/Theme';
import Film from '../Film/Film';


const Films = ({ films, albumClickHandler }) => (
  <FilmsBackground>
    <FilmsWrapper>
      {films.map((film) => <Film info={film} key={film.id} albumClickHandler={albumClickHandler} />)}
    </FilmsWrapper>
  </FilmsBackground>
);

export default Films;

Films.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
  albumClickHandler: PropTypes.func.isRequired,
};

const FilmsBackground = styled.div`
    flex: 1 0 auto;
    width: 100%;    
    background-color: ${Theme.colors.white};  
    padding: 30px 0; 
`;

const FilmsWrapper = styled.div`
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    width: 80%;
    margin: 0 auto;    
`;
