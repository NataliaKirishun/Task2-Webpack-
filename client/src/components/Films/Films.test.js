import React from 'react';
import { render } from 'enzyme';
import Films from './Films';

describe('Films Snapshot', () => {
    it('Films Component should render correctly with given props', () => {
        const  films=[
            {
                "id":5924,
                "title":"Papillon",
                "tagline":"The greatest adventure of escape!",
                "vote_average":7.8,
                "vote_count":564,
                "release_date":"1973-12-13",
                "poster_path":"https://image.tmdb.org/t/p/w500/xY7k2M9eElJnLSYwqY85Xon5z6m.jpg",
                "overview":"A man befriends a fellow criminal as the two of them begin serving their sentence on a dreadful prison island, which inspires the man to plot his escape.",
                "budget":12000000,
                "revenue":53267000,
                "genres":["Crime","Drama"],
                "runtime":151
            },{
                "id":11031,
                "title":"This Is Spinal Tap",
                "tagline":"Does for rock & roll what 'The Sound of Music' did for hills.",
                "vote_average":7.7,
                "vote_count":517,
                "release_date":"1984-05-04",
                "poster_path":"https://image.tmdb.org/t/p/w500/2VDPeoPJ2bKdmfuJxxCktx1hr5g.jpg",
                "overview":"\"This Is Spinal Tap\" shines a light on the self-contained universe of a metal band struggling to get back on the charts, including everything from its complicated history of ups and downs, gold albums, name changes and undersold concert dates, along with the full host of requisite groupies, promoters, hangers-on and historians, sessions, release events and those special behind-the-scenes moments that keep it all real.",
                "budget":2500000,
                "revenue":4736202,
                "genres":["Comedy","Music"],
                "runtime":82
            }
        ];
        const albumClickHandler=jest.fn();
        const component = render(<Films films={films} albumClickHandler={albumClickHandler} />);
        expect(component).toMatchSnapshot();
    });
});

