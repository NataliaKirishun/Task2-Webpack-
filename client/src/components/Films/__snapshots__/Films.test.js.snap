// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films Snapshot Films Component should render correctly with given props 1`] = `
<div
  class="sc-EHOje hkQRmw"
>
  <div
    class="sc-bZQynM gSRbGQ"
  >
    <a
      class="sc-bdVaJa mhsjn"
    >
      <img
        class="sc-bwzfXH ifbPgJ"
        src="https://image.tmdb.org/t/p/w500/xY7k2M9eElJnLSYwqY85Xon5z6m.jpg"
      />
      <div
        class="sc-htpNat eUMQJL"
      >
        <h3
          class="sc-bxivhb gEscEi"
        >
          Papillon
        </h3>
        <span>
          Crime, Drama
        </span>
        <span
          class="sc-ifAKCX cCwMba"
        >
          1973
        </span>
      </div>
    </a>
    <a
      class="sc-bdVaJa mhsjn"
    >
      <img
        class="sc-bwzfXH ifbPgJ"
        src="https://image.tmdb.org/t/p/w500/2VDPeoPJ2bKdmfuJxxCktx1hr5g.jpg"
      />
      <div
        class="sc-htpNat eUMQJL"
      >
        <h3
          class="sc-bxivhb gEscEi"
        >
          This Is Spinal Tap
        </h3>
        <span>
          Comedy, Music
        </span>
        <span
          class="sc-ifAKCX cCwMba"
        >
          1984
        </span>
      </div>
    </a>
  </div>
</div>
`;
